src/app/code/community/Divante/VueStorefrontIndexer/Model/Resource/Catalog/Category.php

```diff
@@ -48,14 +48,25 @@ public function getCategories($storeId = 1, array $categoryIds = [], $fromId = 0
         $rootCategoryId = Mage::app()->getStore($storeId)->getRootCategoryId();
         $rootCategory = Mage::getModel('catalog/category')->load($rootCategoryId);
 
-        $select = $this->connection->select()->from(['e' => $this->coreResource->getTableName('catalog/category')]);
+        $isFlatEnabled = Mage::helper('catalog/category_flat')->isEnabled();
+
+        if($isFlatEnabled) {
+            $tableName = sprintf('catalog_category_flat_store_%d', $storeId);
+            $select = $this->connection->select()->from(['e' => $tableName ]);
+            $select->where('e.include_in_menu = ?', 1);
+        } else {
+            $select = $this->connection->select()->from(['e' => $this->coreResource->getTableName('catalog/category')]);
+        }
 
         if (!empty($categoryIds)) {
             $select->where('e.entity_id IN (?)', $categoryIds);
         }
 
-        $path = "1/{$rootCategory->getId()}%";
-        $select->where('path LIKE ?', $path);
+        /**
+         *  Doesnt work is certain circumstances
+         *  $path = "1/{$rootCategory->getId()}%";
+         *  $select->where('path LIKE ?', $path);
+         */
```

Comment:
Please give an example in which circumstance it does not work? 
Because another way it will export all categories for every store -> which is not the aim here (when using catalog/category table)